barman/retention_policies.py

```diff
@@ -126,6 +126,29 @@ def to_json(self):
         """
         return "%s %s %s" % (self.mode, self.value, self.unit)
 
+    def propagate_retention_status_to_children(self, backup_info, report, ret_status):
+        """
+        Propagate retention status to all backups in the chain.
+
+        .. note::
+            This has a side-effect. It modifies or add data to *report* dict.
+
+        :param barman.infofile.BackupInfo backup_info: The object we want to
+            propagate the RETENTION STATUS from.
+        :param report dict[str, str]: The report data structure to be modified.
+            Each key is the ID of a backup, and its value is the retention status
+            of that backup.
+        :param ret_status str:  The status of the backup according to retention
+            policies
+        """
+        backup_id_tree = backup_info.walk_backups_tree(return_root=False)
+        for backup in backup_id_tree:
```

Comment:
```suggestion
        backup_tree = backup_info.walk_backups_tree(return_root=False)
        for backup in backup_tree:
```

It's not a "backup ID tree" anymore :)